queue_downloads.py

```diff
@@ -0,0 +1,301 @@
+            "Submits DataGateway Download requests for a list of specific filepaths. "
+            "The list will be split into separate parts of up to 10,000 files for "
+            "performance reasons. Once submitted Downloads will be visible in the "
+            "DataGateway UI as usual."
+        ),
+    )
+    parser.add_argument(
+        "input_file",
+        type=str,
+        help=(
+            "File containing the full paths of all files to submit for download, "
+            "separated by newlines. The path should match the 'location' field "
+            "displayed in the DataGateway UI."
+        ),
+    )
+    parser.add_argument(
+        "--url",
+        type=str,
+        default="https://datagateway.diamond.ac.uk",
+        help="The url address of the DataGateway instance to submit requests to.",
+    )
+    parser.add_argument(
+        "-a",
+        "--authenticator",
+        type=str,
+        default="ldap",
+        help="The authentication mechanism to use for DataGateway login.",
+    )
+    parser.add_argument(
+        "-u",
+        "--username",
+        type=str,
+        required=True,
+        help="The username used for DataGateway login.",
+    )
+    parser.add_argument(
+        "-p",
+        "--password-file",
+        type=str,
+        help=(
+            "Location of file containing password for DataGateway login. If not "
+            "provided, the password will need to be provided by prompt."
+        ),
+    )
+    parser.add_argument(
+        "--download-name",
+        type=str,
+        help=(
+            "Custom file name for the download(s). If not set will default to the "
+            "current date and time. '_part_N' will be appended to the each part "
+            "Download of up to 10,000 files."
+        ),
+    )
+    parser.add_argument(
+        "--access-method",
+        type=str,
+        default="dls",
+        choices=["https", "globus", "dls"],
+        help=(
+            "The choice of access method for the data. https: download files via your "
+            "browser. globus: download the data to Globus Online. dls: restore to the "
+            "DLS file system. Data will be available in one of the following directory "
+            "structures: /dls/staging/dls/$instrument/data/$year/$visit or "
+            "/dls/staging/dls/$village/data/$proposal/$visit. Users will have 15 days "
+            "to process and transfer their data. After that period, data will be "
+            "deleted from DLS filesystem."
+        ),
+    )
+    parser.add_argument(
+        "--email-address",
+        type=str,
+        help="Optional address to email status messages to.",
+    )
+    parser.add_argument(
+        "-m",
+        "--monitor-interval",
+        type=float,
+        default=0,
+        help=(
+            "Monitor the submitted downloads to see if they are complete with an "
+            "interval of this many seconds. Non-positive values will disable "
```

Comment:
I feel the interval should be in minutes rather than seconds (even minutes is perhaps a bit optimistic!). Having it in seconds gives the impression that there might be progress to report on that timescale, which is unrealistic.